src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Tables/Instance.sql

```diff
@@ -45,98 +47,84 @@ INCLUDE
 )
 WITH (DATA_COMPRESSION = PAGE)
 
--- Used in WADO and QIDO, putting PartitionKey last allows us to query across partitions in the future.
-CREATE NONCLUSTERED INDEX IX_Instance_StudyInstanceUid_Status_PartitionKey on dbo.Instance
+-- Used in WADO
+CREATE NONCLUSTERED INDEX IX_Instance_PartitionKey_Status_StudyInstanceUid_SeriesInstanceUid_SopInstanceUid on dbo.Instance
 (
-    StudyInstanceUid,
+    PartitionKey,
     Status,
-    PartitionKey    
-)
-INCLUDE
-(
-    SeriesInstanceUid,
-    SopInstanceUid,
-    Watermark,
-    TransferSyntaxUid
-)
-WITH (DATA_COMPRESSION = PAGE)
-
--- Used in WADO and QIDO, putting PartitionKey last allows us to query across partitions in the future.
-CREATE NONCLUSTERED INDEX IX_Instance_StudyInstanceUid_SeriesInstanceUid_Status_PartitionKey on dbo.Instance
-(
     StudyInstanceUid,
     SeriesInstanceUid,
-    Status,
-    PartitionKey    
+    SopInstanceUid
 )
 INCLUDE
 (
-    SopInstanceUid,
     Watermark,
-    TransferSyntaxUid
+    TransferSyntaxUid,
+    HasFrameMetadata
 )
 WITH (DATA_COMPRESSION = PAGE)
 
--- Used in WADO and QIDO, putting PartitionKey last allows us to query across partitions in the future.
-CREATE NONCLUSTERED INDEX IX_Instance_SopInstanceUid_Status_PartitionKey on dbo.Instance
+-- Used in GetInstancesByWatermarkRange
+CREATE UNIQUE NONCLUSTERED INDEX IX_Instance_Watermark_Status on dbo.Instance
 (
-    SopInstanceUid,
-    Status,
-    PartitionKey    
+    Watermark,
+    Status
 )
 INCLUDE
 (
+    PartitionKey,
     StudyInstanceUid,
     SeriesInstanceUid,
-    Watermark
+    SopInstanceUid
 )
 WITH (DATA_COMPRESSION = PAGE)
 
--- Used in GetInstancesByWatermarkRange
-CREATE UNIQUE NONCLUSTERED INDEX IX_Instance_Watermark_Status on dbo.Instance
+-- QIDO filtering
```

Comment:
There is one more case where we get instances within a study. we may need a new index with 

```
(
    PartitionKey,
    StudyInstanceUid,
    SopInstanceUid
)
```

```
/*
This query text was retrieved from showplan XML, and may be truncated.
*/

SELECT
      f.StudyInstanceUid,
      f.SeriesInstanceUid,
      f.SopInstanceUid,
      f.Watermark
      FROM
      ( SELECT
      i.StudyInstanceUid
      ,i.SeriesInstanceUid
      ,i.SopInstanceUid
      ,i.Watermark
      FROM dbo.Study st
      INNER JOIN dbo.Series se
      ON se.StudyKey = st.StudyKey
      AND se.PartitionKey = st.PartitionKey
      INNER JOIN dbo.Instance i
      ON i.SeriesKey = se.SeriesKey
      AND i.PartitionKey = se.PartitionKey
      AND i.Status = 1
      WHERE 1 = 1
      AND st.PartitionKey = 1
      AND i.SopInstanceUid=''
      AND st.StudyInstanceUid=''
      ORDER BY i.Watermark DESC
      OFFSET 0 ROWS
      FETCH NEXT 100 ROWS ONLY
      ) f
      OPTION(RECOMPILE)



```

Reply:
They are on different tables. st is PartitionKey and StudyInstanceUid. Instance SopInstanceUid.

Reply:
My bad, I was trying something and forget to add as comment, when users are requesting only instance level data, we can totally avoid joining series and study, we could return all the values from Instance table itself. If that is the case, we may need a index with StudyInstanceUID. What do you think about this thought?

Reply:
Yes, we can definetly do it in the future. it complicated the query generation, we can do it later. I am expecting very few instance level queries. I have seen a lot of study and few series level queries

Reply:
Makes sense